NewCalculatorS/NewCalculator/BracketsOpener.cpp

```diff
@@ -10,11 +10,23 @@ BracketsOpener::~BracketsOpener()
     //dtor
 }
 
-string BracketsOpener::remove_brackets(string s, int start, int finish, string number)
+string BracketsOpener::remove_brackets(string s, int *start, int finish, string number)
 {
-    if ((start > 0) && (s[start - 1] == OPENING_BRACKET))
-        return s.replace(start - 1, finish - start + 3, number);
-    return s.replace(start, finish - start + 1, number);
+    if ((*start > 0) && (s[(*start) - 1] == OPENING_BRACKET))
+    {
+        if (((*start) > 1) && (s[(*start) - 2] > 97))
```

Comment:
97 is a **magic number**.
https://dev.to/10xlearner/magic-numbers-and-how-to-deal-with-them-in-c-2jbn